packages/cli/src/commands/serve.ts

```diff
@@ -27,6 +31,11 @@ class PlatformAppDevelopmentServer {
             });
 
             const server = await viteServer.listen(this.port, true);
+
+            if (viteServer.config.server?.port && viteServer.config.server?.port !== this.port) {
```

Comment:
Can you instead let Vite tell the user the port is already used by setting [`strictPort`](https://vitejs.dev/config/server-options.html#server-strictport)?